shared/common-adapters/markdown.js

```diff
@@ -0,0 +1,118 @@
+// @flow
+
+import Text from './text'
+import React from 'react'
+import {List} from 'immutable'
+
+import type {Props} from './markdown'
+import type {PropsOf} from '../constants/types/more'
+
+// Order matters, since we want to match the longer ticks first
+const openToClosePair = {
+  '```': '```',
+  '`': '`',
+  '*': '*',
+  '_': '_',
+  '~': '~',
+}
+
+type TagInfo<C> = {Component: Class<C>, props: PropsOf<C>}
+
+const initialOpenToTag = {
+  '`': {Component: Text, props: {type: 'CodeSnippet', style: {}}},
+  '```': {Component: Text, props: {type: 'CodeSnippetBlock'}},
+  '*': {Component: Text, props: {type: 'BodySemibold'}},
+  '_': {Component: Text, props: {type: 'BodyInherit', style: {fontStyle: 'italic'}}},
+  '~': {Component: Text, props: {type: 'BodyInherit', style: {textDecoration: 'line-through'}}},
+}
+
+const openToNextOpenToTag = {
```

Comment:
The special tags you look for are dependent on what tags you are in. In code blocks you don't care if you see a `*`, so this matches that behavior by putting the opening mark to tag data in the stack structure of the parser